src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1116,15 +1117,25 @@ private void visitVarDeclaration(String name, JSType type) {
 
     private void visitTemplateTypes(ObjectType type) {
       if (type.hasAnyTemplateTypes() && !type.getTemplateTypeMap().isEmpty()) {
-        emit("<");
-        Iterator<TemplateType> it = type.getTemplateTypeMap().getTemplateKeys().iterator();
-        while (it.hasNext()) {
-          emit(it.next().getDisplayName());
-          if (it.hasNext()) {
-            emit(",");
+        // Closure compiler add a fake template types like IObject#KEY or IObject#Value
+        // for several types. The following code is a workaround waiting this bug is fixed in the
+        // closure compiler
+        // TODO(dramaix): remove this code when the bug is fixed in JsCompiler.
+        List<String> realTemplateType = new ArrayList<>();
+
+        for (TemplateType templateType: type.getTemplateTypeMap().getTemplateKeys()) {
+          String displayName = templateType.getDisplayName();
+
+          if (displayName.indexOf('#') < 0) {
```

Comment:
I have run into this issue before, I think it is safe to just strip the part before `#` , instead of complete omitting it.
 So the TS type will become `Object<Key, Value>`.